flocker/node/agents/functional/test_cinder_behaviour.py

```diff
@@ -112,3 +115,69 @@ def test_updated_metadata_is_listed(self):
         missing_items = expected_items - actual_items
 
         self.assertEqual(set(), missing_items)
+
+
+class FakeTime(object):
+    def __init__(self, initial_time):
+        self._current_time = initial_time
+
+    def time(self):
+        return self._current_time
+
+    def sleep(self, interval):
+        self._current_time += interval
+
+
+class BlockDeviceAPIDestroyTests(SynchronousTestCase):
+    """
+    Test for ``cinder.CinderBlockDeviceAPI.1volume``
+    """
+    def setUp(self):
+        self.cinder_volumes = cinder_volume_manager()
+
+    def test_destroy_timesout(self):
+        """
+        If the cinder cannot delete the volume, we should timeout
+        after waiting some time
+        """
+        new_volume = self.cinder_volumes.create(size=100)
+        self.addCleanup(self.cinder_volumes.delete, new_volume)
+        listed_volume = wait_for_volume_state(
+            volume_manager=self.cinder_volumes,
+            expected_volume=new_volume,
+            desired_state=u'available',
+            transient_states=(u'creating',),
+        )
+        expected_timeout = 8
+        # Using a fake no-op delete so it doesn't actually delete anything
+        # (we don't need any actual volumes here, as we only need to verify
+        # the timeout)
+        self.patch(self.cinder_volumes, "delete", lambda *args, **kwargs: None)
+        # Now try to delete it
+        time_module = FakeTime(initial_time=0)
+        api = CinderBlockDeviceAPI(
+            cinder_volume_manager=self.cinder_volumes,
+            nova_volume_manager=object(),
+            nova_server_manager=object(),
+            cluster_id=uuid4(),
+            # Setting the timeut to 1, as the default is quite high,
```

Comment:
No longer relevant comment. In fact you don't need that argument at all given fake time module I guess?